src/AzureExtension/Providers/RepositoryProvider.cs

```diff
@@ -394,4 +357,322 @@ public void Dispose()
     {
         GC.SuppressFinalize(this);
     }
+
+    public IAsyncOperation<RepositoriesResult> GetRepositoriesAsync(IReadOnlyDictionary<string, string> searchTerms, IDeveloperId developerId)
+    {
+        // First query will find repos only if the user made a PR.
+        var shouldIgnorePRDateOld = _shouldIgnorePRDate;
+        _shouldIgnorePRDate = true;
+        return Task.Run(() =>
+        {
+            try
+            {
+                // Store the search terms.
+                var serverToUse = searchTerms.ContainsKey(_server) ? searchTerms[_server] : string.Empty;
+                _selectedServer = serverToUse;
+
+                var organizationToUse = searchTerms.ContainsKey(_organization) ? searchTerms[_organization] : string.Empty;
+                _selectedOrganization = organizationToUse;
+
+                var projectToUse = searchTerms.ContainsKey(_project) ? searchTerms[_project] : string.Empty;
+                _selectedProject = projectToUse;
+
+                // Get access token for ADO API calls.
+                if (developerId is not DeveloperId.DeveloperId azureDeveloperId)
+                {
+                    var exception = new NotSupportedException("Authenticated user is not the an azure developer id.");
+                    return new RepositoriesResult(exception, $"{exception.Message} HResult: {exception.HResult}");
+                }
+
+                _azureHierarchy ??= new AzureRepositoryHierarchy(azureDeveloperId);
+
+#pragma warning disable CA1309 // Use ordinal string comparison.  Organization names should match exactly.
+                var organizations = _azureHierarchy.GetOrganizationsAsync().Result;
+                var orgsInModernUrlFormat = new List<Organization>();
+                if (_selectedServer.Equals(_defaultSearchServerName))

```

Comment:
do we need any special comparison logic in this Equals call?

Reply:
Because the extension is comparing strings made by humans I went with a linguistic comparison over a bit-wise comparison.